$NetBSD: patch-bh,v 1.6 2011/09/18 01:36:06 manu Exp $

NetBSD's <dirent.h> does a #defiine d_ino d_fileno. Since glusterfs 
declares structs with d_ino members, we must include <dirent.h> early
so that d_ino get preprocessed into d_fileno conssitently.

Also define proper xdr types for NetBSD.
--- rpc/rpc-lib/src/xdr-common.h.orig	2011-08-23 14:31:38.000000000 +0200
+++ rpc/rpc-lib/src/xdr-common.h	2011-09-17 15:45:29.000000000 +0200
@@ -29,8 +29,13 @@
 #include <sys/types.h>
 #include <rpc/xdr.h>
 #include <sys/uio.h>
 
+#ifdef __NetBSD__
+#include <dirent.h>
+#endif /* __NetBSD__ */
+
+
 enum gf_dump_procnum {
         GF_DUMP_NULL,
         GF_DUMP_DUMP,
         GF_DUMP_MAXVALUE,
@@ -46,8 +51,15 @@
 #define xdr_uint32_t xdr_u_int32_t
 #define uint64_t u_int64_t
 #endif
 
+#if defined(__NetBSD__)
+#define xdr_u_quad_t xdr_u_int64_t
+#define xdr_quad_t   xdr_int64_t
+#define xdr_uint32_t xdr_u_int32_t
+#endif
+
+
 #if GF_SOLARIS_HOST_OS
 #define u_quad_t uint64_t
 #define quad_t int64_t
 #define xdr_u_quad_t xdr_uint64_t
